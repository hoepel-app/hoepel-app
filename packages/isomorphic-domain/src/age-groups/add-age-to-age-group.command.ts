import { Command } from '../command/command'

export type AddAgeToAgeGroupCommandProps = {
  tenantId: string
  ageGroupName: string
  age: number
}

export class AddAgeToAgeGroupCommand
  implements Command<AddAgeToAgeGroupCommandProps> {
  name = 'add-age-to-age-group-command' as const

  private constructor(private readonly props: AddAgeToAgeGroupCommandProps) {}

  static create(
    tenantId: string,
    ageGroupName: string,
    age: number
  ): AddAgeToAgeGroupCommand {
    return new AddAgeToAgeGroupCommand({
      tenantId,
      age,
      ageGroupName,
    })
  }

  toProps(): AddAgeToAgeGroupCommandProps {
    return this.props
  }

  get tenantId(): string {
    return this.props.tenantId
  }

  get ageGroupName(): string {
    return this.props.ageGroupName
  }

  get age(): number {
    return this.props.age
  }
}
