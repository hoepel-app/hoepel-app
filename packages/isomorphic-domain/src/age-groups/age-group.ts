import { DayDate } from '@hoepel.app/types'
import {
  differenceInCalendarYears,
  differenceInDays,
  setYear,
  getYear,
} from 'date-fns'

export type SwitchOverOn = 'childs-birthday' | 'new-school-year'

export type AgeGroupProps = {
  name: string
  validForAges: readonly number[]
}

export class AgeGroup {
  private constructor(private readonly props: AgeGroupProps) {}

  belongsToThisGroup(
    birthDate: DayDate,
    switchOverOn: SwitchOverOn,
    currentDate: DayDate = DayDate.today()
  ): boolean {
    if (switchOverOn === 'childs-birthday') {
      const currentAge = differenceInCalendarYears(
        currentDate.nativeDate,
        birthDate.nativeDate
      )

      return this.validForAges.has(currentAge)
    } else {
      const newSchoolYear = this.dateIsInNewSchoolYear(currentDate)
      const hasHadBirthDay = this.hasHadBirthDayThisYear(birthDate, currentDate)

      const currentAge = differenceInCalendarYears(
        currentDate.nativeDate,
        birthDate.nativeDate
      )

      if (!newSchoolYear && hasHadBirthDay) {
        return this.validForAges.has(currentAge - 1)
      } else {
        return this.validForAges.has(currentAge)
      }
    }
  }

  static create(name: string, validForAges: ReadonlySet<number>): AgeGroup {
    return AgeGroup.fromProps({
      name,
      validForAges: [...validForAges],
    })
  }

  static fromProps(props: AgeGroupProps): AgeGroup {
    return new AgeGroup(props)
  }

  toProps(): AgeGroupProps {
    return this.props
  }

  withAgeAdded(age: number): AgeGroup {
    return AgeGroup.fromProps({
      ...this.toProps(),
      validForAges: [...new Set([...this.validForAges, age])],
    })
  }

  withAgeRemoved(age: number): AgeGroup {
    return AgeGroup.fromProps({
      ...this.toProps(),
      validForAges: [...this.validForAges].filter((a) => a !== age),
    })
  }

  withName(name: string): AgeGroup {
    return AgeGroup.fromProps({
      ...this.toProps(),
      name,
    })
  }

  agesOverlapWith(other: AgeGroup): ReadonlySet<number> {
    const otherAges = [...other.validForAges]

    return new Set(
      [...this.validForAges].filter((age) => otherAges.includes(age))
    )
  }

  private hasHadBirthDayThisYear(
    birthDate: DayDate,
    currentDate: DayDate
  ): boolean {
    const daysSinceBirthday = differenceInDays(
      currentDate.nativeDate,
      setYear(birthDate.nativeDate, getYear(currentDate.nativeDate))
    )

    return daysSinceBirthday >= 0
  }

  private dateIsInNewSchoolYear(date: DayDate): boolean {
    return date.month >= 9
  }

  get name(): string {
    return this.props.name
  }

  get validForAges(): Set<number> {
    return new Set(this.props.validForAges)
  }
}
