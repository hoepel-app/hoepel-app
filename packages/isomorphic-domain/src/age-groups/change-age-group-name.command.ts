import { Command } from '../command/command'

export type ChangeAgeGroupNameCommandProps = {
  tenantId: string
  ageGroupCurrentName: string
  ageGroupNewName: string
}

export class ChangeAgeGroupNameCommand
  implements Command<ChangeAgeGroupNameCommandProps> {
  name = 'change-age-group-name-command' as const

  private constructor(private readonly props: ChangeAgeGroupNameCommandProps) {}

  static create(
    tenantId: string,
    ageGroupCurrentName: string,
    ageGroupNewName: string
  ): ChangeAgeGroupNameCommand {
    return new ChangeAgeGroupNameCommand({
      tenantId,
      ageGroupCurrentName,
      ageGroupNewName,
    })
  }

  toProps(): ChangeAgeGroupNameCommandProps {
    return this.props
  }

  get tenantId(): string {
    return this.props.tenantId
  }

  get ageGroupCurrentName(): string {
    return this.props.ageGroupCurrentName
  }

  get ageGroupNewName(): string {
    return this.props.ageGroupNewName
  }
}
