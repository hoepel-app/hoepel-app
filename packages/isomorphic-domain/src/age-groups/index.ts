export { AgeGroupsApplicationService } from './age-groups.application-service'
export { ChangeSwitchOverOnCommand } from './change-switch-over-on.command'
export { RemoveAgeGroupCommand } from './remove-age-group.command'
export { AddAgeGroupCommand } from './add-age-group.command'
export { AgeGroup, SwitchOverOn } from './age-group'
export { AgeGroups, AgeGroupsProps } from './age-groups'
export { AgeGroupsRepository } from './age-groups.repository'
