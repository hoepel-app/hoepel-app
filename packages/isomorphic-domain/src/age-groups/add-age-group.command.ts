import { Command } from '../command/command'
import { AgeGroupProps, AgeGroup } from './age-group'

export type AddAgeGroupCommandProps = {
  tenantId: string
  ageGroupProps: AgeGroupProps
}

export class AddAgeGroupCommand implements Command<AddAgeGroupCommandProps> {
  private constructor(private readonly props: AddAgeGroupCommandProps) {}

  static create(tenantId: string, ageGroup: AgeGroup): AddAgeGroupCommand {
    return new AddAgeGroupCommand({
      tenantId,
      ageGroupProps: ageGroup.toProps(),
    })
  }

  toProps(): AddAgeGroupCommandProps {
    return this.props
  }

  get tenantId(): string {
    return this.props.tenantId
  }

  get ageGroup(): AgeGroup {
    return AgeGroup.fromProps(this.props.ageGroupProps)
  }
}
