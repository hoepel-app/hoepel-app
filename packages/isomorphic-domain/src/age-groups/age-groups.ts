import { AgeGroupProps, AgeGroup, SwitchOverOn } from './age-group'
import { DayDate } from '@hoepel.app/types'

export type AgeGroupsProps = {
  ageGroups: readonly AgeGroupProps[]
  switchOverOn: SwitchOverOn
}
export class AgeGroups {
  private constructor(private readonly props: AgeGroupsProps) {}

  get ageGroups(): readonly AgeGroup[] {
    return this.props.ageGroups.map((ageGroup) => AgeGroup.fromProps(ageGroup))
  }

  get names(): ReadonlySet<string> {
    return new Set(this.ageGroups.map((group) => group.name))
  }

  get switchOverOn(): SwitchOverOn {
    return this.props.switchOverOn
  }

  /** Ages that are not in use yet by any age group */
  get unusedAges(): ReadonlySet<number> {
    const usedAges = this.usedAges
    return new Set(
      [...AgeGroups.validAges.values()].filter((age) => !usedAges.has(age))
    )
  }

  get usedAges(): ReadonlySet<number> {
    const ages = this.ageGroups.map((group) => group.validForAges)

    return ages.reduce(
      (acc, curr) => new Set([...acc, ...curr]),
      new Set<number>()
    )
  }

  static fromProps(props: AgeGroupsProps): AgeGroups {
    return new AgeGroups(props)
  }

  static create(switchOverOn: SwitchOverOn): AgeGroups {
    return this.fromProps({
      switchOverOn,
      ageGroups: [],
    })
  }

  static createEmpty(): AgeGroups {
    return this.create('childs-birthday')
  }

  static get validAges(): ReadonlySet<number> {
    return new Set([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16])
  }

  toProps(): AgeGroupsProps {
    return {
      switchOverOn: this.switchOverOn,
      ageGroups: this.ageGroups.map((group) => group.toProps()),
    }
  }

  withAddedAgeGroup(ageGroup: AgeGroup): AgeGroups {
    return AgeGroups.fromProps({
      ...this.toProps(),
      ageGroups: [...this.toProps().ageGroups, ageGroup.toProps()],
    })
  }

  withSwitchOverOn(switchOverOn: SwitchOverOn): AgeGroups {
    return AgeGroups.fromProps({
      ...this.toProps(),
      switchOverOn,
    })
  }

  withAgeGroupRemoved(ageGroupName: string): AgeGroups {
    return AgeGroups.fromProps({
      ...this.toProps(),
      ageGroups: this.ageGroups
        .filter((group) => group.name !== ageGroupName)
        .map((group) => group.toProps()),
    })
  }

  classifyPerson(
    birthDate: DayDate,
    currentDate: DayDate = DayDate.today()
  ): AgeGroup | null {
    return (
      this.ageGroups.find((ageGroup) =>
        ageGroup.belongsToThisGroup(birthDate, this.switchOverOn, currentDate)
      ) || null
    )
  }

  mayAddAgeGroup(ageGroup: AgeGroup): boolean {
    const overlap = this.ageGroups
      .map((group) => group.agesOverlapWith(ageGroup))
      .reduce((acc, set) => new Set([...acc, ...set]), new Set())

    const nameAlreadyExists = this.names.has(ageGroup.name)

    return overlap.size === 0 && !nameAlreadyExists
  }

  findAgeGroupWithName(name: string): AgeGroup | null {
    return this.ageGroups.find((group) => group.name === name) || null
  }
}
