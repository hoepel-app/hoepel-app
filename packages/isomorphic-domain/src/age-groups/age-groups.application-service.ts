import { AgeGroupsRepository } from './age-groups.repository'
import { CommandResult } from '@hoepel.app/ddd-library'
import { AddAgeGroupCommand } from './commands/add-age-group.command'
import { ChangeSwitchOverOnCommand } from './commands/change-switch-over-on.command'
import { RemoveAgeGroupCommand } from './commands/remove-age-group.command'
import { AddAgeToAgeGroupCommand } from './commands/add-age-to-age-group.command'
import { AgeGroups } from './age-groups'
import { Observable } from 'rxjs'
import { first } from 'rxjs/operators'
import { RemoveAgeFromAgeGroupCommand } from './commands/remove-age-from-age-group.command'
import { ChangeAgeGroupNameCommand } from './commands/change-age-group-name.command'

export class AgeGroupsApplicationService {
  constructor(private readonly repo: AgeGroupsRepository) {}

  findAgeGroups(tenantId: string): Observable<AgeGroups> {
    return this.repo.getForTenant(tenantId)
  }

  async addAgeGroup(command: AddAgeGroupCommand): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    if (!ageGroups.mayAddAgeGroup(command.ageGroup)) {
      return { status: 'rejected' }
    }

    const newAgeGroups = ageGroups.withAddedAgeGroup(command.ageGroup)

    await this.repo.put(newAgeGroups)

    return { status: 'accepted' }
  }

  async changeSwitchOverOn(
    command: ChangeSwitchOverOnCommand
  ): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(ageGroups.withSwitchOverOn(command.switchOverOn))

    return { status: 'accepted' }
  }

  async removeAgeGroup(command: RemoveAgeGroupCommand): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(ageGroups.withAgeGroupRemoved(command.ageGroupName))

    return { status: 'accepted' }
  }

  async removeAgeFromAgeGroup(
    command: RemoveAgeFromAgeGroupCommand
  ): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(
      ageGroups.withAgeRemovedFromAgeGroup(command.ageGroupName, command.age)
    )

    return { status: 'accepted' }
  }

  async addAgeToAgeGroup(
    command: AddAgeToAgeGroupCommand
  ): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(
      ageGroups.withAgeAddedToAgeGroup(command.ageGroupName, command.age)
    )

    return { status: 'accepted' }
  }

  async changeAgeGroupName(
    command: ChangeAgeGroupNameCommand
  ): Promise<CommandResult> {
    const ageGroups = await this.repo
      .getForTenant(command.tenantId)
      .pipe(first())
      .toPromise()

    await this.repo.put(
      ageGroups.withAgeGroupRenamed(
        command.ageGroupCurrentName,
        command.ageGroupNewName
      )
    )

    return { status: 'accepted' }
  }
}
