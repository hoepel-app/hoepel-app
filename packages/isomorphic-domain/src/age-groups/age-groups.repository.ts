import { AgeGroups } from './age-groups'
import { Observable } from 'rxjs'

export type AgeGroupsRepository = {
  findForTenant(tenantId: string): Observable<AgeGroups>
  putForTenant(tenantId: string, entity: AgeGroups): Promise<void>
}
