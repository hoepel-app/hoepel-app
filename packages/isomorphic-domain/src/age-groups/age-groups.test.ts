import { AgeGroups } from './age-groups'
import { DayDate } from '@hoepel.app/types'
import { AgeGroup } from './age-group'

describe('Age groups based on current age', () => {
  const ageGroups = AgeGroups.create('childs-birthday')
    .withAddedAgeGroup(AgeGroup.create('Kleuters', new Set([2, 3, 4])))
    .withAddedAgeGroup(AgeGroup.create('Mini', new Set([5, 6, 7, 8])))
    .withAddedAgeGroup(AgeGroup.create('Maxi', new Set([9, 10])))
    .withAddedAgeGroup(AgeGroup.create('Tieners', new Set([11, 12, 13])))

  it('classifies a person as belonging according to their current age', () => {
    expect(
      ageGroups.classifyPerson(
        DayDate.fromISO8601('2018-06-05'),
        DayDate.fromISO8601('2020-08-09')
      )?.name
    ).toEqual('Kleuters')
  })

  it('classifies a person as not belonging according to their current age', () => {
    const ageGroup = ageGroups.classifyPerson(
      DayDate.fromISO8601('2008-06-05'),
      DayDate.fromISO8601('2020-08-09')
    )

    expect(ageGroup?.name).not.toEqual('Kleuters')
    expect(ageGroup?.name).not.toEqual('Mini')
  })

  it('classifies a person as not in the age group if today is their birthday and they are older', () => {
    const ageGroup = ageGroups.classifyPerson(
      DayDate.fromISO8601('2015-06-13'),
      DayDate.fromISO8601('2020-06-13')
    )

    expect(ageGroup?.name).not.toEqual('Kleuters')
    expect(ageGroup?.name).toEqual('Mini')
  })

  it('returns null if no age groups match', () => {
    const ageGroup = ageGroups.classifyPerson(
      DayDate.fromISO8601('2019-06-13'),
      DayDate.fromISO8601('2020-06-13')
    )

    expect(ageGroup).toBeNull()
  })

  it('classifies a person as in the age group if today is their birthday and they are just old enough', () => {
    expect(
      ageGroups.classifyPerson(
        DayDate.fromISO8601('2010-06-13'),
        DayDate.fromISO8601('2020-06-13')
      )?.name
    ).toEqual('Maxi')
  })
})

describe('Age groups switching only when a new school year starts', () => {
  const ageGroups = AgeGroups.create('new-school-year')
    .withAddedAgeGroup(AgeGroup.create('Kleuters', new Set([2, 3, 4])))
    .withAddedAgeGroup(AgeGroup.create('Mini', new Set([5, 6, 7, 8])))
    .withAddedAgeGroup(AgeGroup.create('Maxi', new Set([9, 10])))
    .withAddedAgeGroup(AgeGroup.create('Tieners', new Set([11, 12, 13])))

  it('classifies a person according to their current age ', () => {
    expect(
      ageGroups.classifyPerson(
        DayDate.fromISO8601('2017-06-05'),
        DayDate.fromISO8601('2020-08-09')
      )?.name
    ).toEqual('Kleuters')
  })

  it('classifies a person in the age group, even though they are older, when the school year has not yet started', () => {
    expect(
      ageGroups.classifyPerson(
        DayDate.fromISO8601('2016-02-13'),
        DayDate.fromISO8601('2020-06-13')
      )?.name
    ).toEqual('Kleuters')
  })

  it('classifies an old enough person as not in the age group when the school year has not yet started', () => {
    const ageGroup = ageGroups.classifyPerson(
      DayDate.fromISO8601('2009-02-13'),
      DayDate.fromISO8601('2020-06-13')
    )

    expect(ageGroup?.name).not.toEqual('Tieners')
    expect(ageGroup?.name).toEqual('Maxi')
  })

  it('classifies an old enough person as in the age group when the school year has started', () => {
    const ageGroup = ageGroups.classifyPerson(
      DayDate.fromISO8601('2009-02-13'),
      DayDate.fromISO8601('2020-09-13')
    )

    expect(ageGroup?.name).toEqual('Tieners')
    expect(ageGroup?.name).not.toEqual('Maxi')
  })
})

describe('AgeGroups', () => {
  const exampleGroups = AgeGroups.create('childs-birthday')
    .withAddedAgeGroup(AgeGroup.create('Kleuters', new Set([2, 3, 4])))
    .withAddedAgeGroup(AgeGroup.create('Mini', new Set([5, 6, 7, 8])))
    .withAddedAgeGroup(AgeGroup.create('Maxi', new Set([9, 10])))
    .withAddedAgeGroup(AgeGroup.create('Tieners', new Set([11, 12, 13])))

  describe('mayAddAgeGroup', () => {
    it('false when age group with name already exists', () => {
      expect(
        exampleGroups.mayAddAgeGroup(AgeGroup.create('Tieners', new Set([14])))
      ).toEqual(false)
    })

    it('false when age group overlaps with existing age group', () => {
      expect(
        exampleGroups.mayAddAgeGroup(AgeGroup.create('Super', new Set([10])))
      ).toEqual(false)
    })

    it("true when ages don't overlap and name doesn't exist yet", () => {
      expect(
        exampleGroups.mayAddAgeGroup(AgeGroup.create('Super', new Set([15])))
      ).toEqual(true)
    })
  })

  describe('findAgeGroupWithName', () => {
    it('returns null if no age group with given name can be found', async () => {
      const group = exampleGroups.findAgeGroupWithName('Middengroep')

      expect(group).toBeNull()
    })

    it('returns age group with given name', async () => {
      const group = exampleGroups.findAgeGroupWithName('Maxi')

      expect(group).toEqual(
        exampleGroups.ageGroups.find((group) => group.name === 'Maxi')
      )
    })
  })

  it('serializing and deserializing yields same object', () => {
    expect(AgeGroups.fromProps(exampleGroups.toProps())).toEqual(exampleGroups)
  })

  describe('usedAges', () => {
    it('be empty when there are no age groups', () => {
      const groups = AgeGroups.create('childs-birthday')

      expect(groups.usedAges).toEqual(new Set())
    })

    it('return a set of used ages', () => {
      expect(exampleGroups.usedAges).toEqual(
        new Set([2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13])
      )
    })
  })

  describe('unusedAges', () => {
    it('be empty when there are no age groups', () => {
      const groups = AgeGroups.create('childs-birthday')

      expect(groups.unusedAges).toEqual(
        new Set([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16])
      )
    })

    it('return a set of unused ages', () => {
      expect(exampleGroups.unusedAges).toEqual(new Set([1, 14, 15, 16]))
    })
  })

  describe('withAgeAddedToAgeGroup', () => {
    it('adds age to existing age group', () => {
      expect(
        exampleGroups.withAgeAddedToAgeGroup('Tieners', 14)
      ).toMatchSnapshot()
    })

    it('does nothing for non-existing age group', () => {
      expect(exampleGroups.withAgeAddedToAgeGroup('non-existant', 14)).toEqual(
        exampleGroups
      )
    })
  })

  describe('withAgeRemovedFromAgeGroup', () => {
    it('removess age to existing age group', () => {
      expect(
        exampleGroups.withAgeRemovedFromAgeGroup('Tieners', 13)
      ).toMatchSnapshot()
    })

    it('does nothing for non-existing age group', () => {
      expect(
        exampleGroups.withAgeRemovedFromAgeGroup('non-existant', 10)
      ).toEqual(exampleGroups)
    })
  })
})
