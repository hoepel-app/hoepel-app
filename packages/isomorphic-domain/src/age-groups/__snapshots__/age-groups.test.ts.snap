// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgeGroups withAgeAddedToAgeGroup adds age to existing age group 1`] = `
AgeGroups {
  "props": Object {
    "ageGroups": Array [
      Object {
        "name": "Kleuters",
        "validForAges": Array [
          2,
          3,
          4,
        ],
      },
      Object {
        "name": "Maxi",
        "validForAges": Array [
          9,
          10,
        ],
      },
      Object {
        "name": "Mini",
        "validForAges": Array [
          5,
          6,
          7,
          8,
        ],
      },
      Object {
        "name": "Tieners",
        "validForAges": Array [
          11,
          12,
          13,
          14,
        ],
      },
    ],
    "switchOverOn": "childs-birthday",
    "tenantId": "my-tenant-id",
  },
}
`;

exports[`AgeGroups withAgeGroupRenamed renames an age group 1`] = `
AgeGroups {
  "props": Object {
    "ageGroups": Array [
      Object {
        "name": "Kleuters",
        "validForAges": Array [
          2,
          3,
          4,
        ],
      },
      Object {
        "name": "Maxi",
        "validForAges": Array [
          9,
          10,
        ],
      },
      Object {
        "name": "Mini",
        "validForAges": Array [
          5,
          6,
          7,
          8,
        ],
      },
      Object {
        "name": "Oudere kinderen",
        "validForAges": Array [
          11,
          12,
          13,
        ],
      },
    ],
    "switchOverOn": "childs-birthday",
    "tenantId": "my-tenant-id",
  },
}
`;

exports[`AgeGroups withAgeRemovedFromAgeGroup removes age from existing age group 1`] = `
AgeGroups {
  "props": Object {
    "ageGroups": Array [
      Object {
        "name": "Kleuters",
        "validForAges": Array [
          2,
          3,
          4,
        ],
      },
      Object {
        "name": "Maxi",
        "validForAges": Array [
          9,
          10,
        ],
      },
      Object {
        "name": "Mini",
        "validForAges": Array [
          5,
          6,
          7,
          8,
        ],
      },
      Object {
        "name": "Tieners",
        "validForAges": Array [
          11,
          12,
        ],
      },
    ],
    "switchOverOn": "childs-birthday",
    "tenantId": "my-tenant-id",
  },
}
`;
